File: SDKs/amazonaws.com/fis/2020-12-01/typescript/src/sdk/models/shared/experiment.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ExperimentAction } from "./experimentaction";
import { ExperimentState } from "./experimentstate";
import { ExperimentStopCondition } from "./experimentstopcondition";
import { ExperimentTarget } from "./experimenttarget";



// Experiment
/** 
 * Describes an experiment.
**/
export class Experiment extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=actions", elemType: ExperimentAction })
  actions?: Record<string, ExperimentAction>;

  @SpeakeasyMetadata({ data: "json, name=creationTime" })
  creationTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=endTime" })
  endTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=experimentTemplateId" })
  experimentTemplateId?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=roleArn" })
  roleArn?: string;

  @SpeakeasyMetadata({ data: "json, name=startTime" })
  startTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=state" })
  state?: ExperimentState;

  @SpeakeasyMetadata({ data: "json, name=stopConditions", elemType: ExperimentStopCondition })
  stopConditions?: ExperimentStopCondition[];

  @SpeakeasyMetadata({ data: "json, name=tags" })
  tags?: Record<string, string>;

  @SpeakeasyMetadata({ data: "json, name=targets", elemType: ExperimentTarget })
  targets?: Record<string, ExperimentTarget>;
}

File: SDKs/amazonaws.com/fis/2020-12-01/python/src/sdk/models/shared/experiment.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import experimentaction as shared_experimentaction
from ..shared import experimentstate as shared_experimentstate
from ..shared import experimentstopcondition as shared_experimentstopcondition
from ..shared import experimenttarget as shared_experimenttarget


@dataclass_json
@dataclasses.dataclass
class Experiment:
    r"""Experiment
    Describes an experiment.
    """
    
    actions: Optional[dict[str, shared_experimentaction.ExperimentAction]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('actions') }})
    creation_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('creationTime'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    end_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('endTime'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    experiment_template_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('experimentTemplateId') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    role_arn: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('roleArn') }})
    start_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('startTime'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    state: Optional[shared_experimentstate.ExperimentState] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('state') }})
    stop_conditions: Optional[list[shared_experimentstopcondition.ExperimentStopCondition]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('stopConditions') }})
    tags: Optional[dict[str, str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tags') }})
    targets: Optional[dict[str, shared_experimenttarget.ExperimentTarget]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('targets') }})
    
